qa/workunits/fs/fscrypt.sh

```diff
@@ -2,118 +2,83 @@
-	rm -rf $testdir
-}
+# Ensure fscrypt CLI is installed and in PATH
+FSCRYPT_CLI="$(type -P fscrypt)"
+if [ -z "$FSCRYPT_CLI" ]; then
+    echo "fscrypt CLI not found after setup. Ensure it was installed correctly."
+    exit 1
+fi
 
-# For now will test the V2 encryption policy only as the
-# V1 encryption policy is deprecated
+# Initialize global fscrypt config
+sudo $FSCRYPT_CLI setup --force --verbose --all-users
 
-install_xfsprogs
-setup_xfstests_env
+MOUNT_POINT=$(dirname "$(dirname "$(pwd)")")
+# Verify that the mount point exists
+if [ ! -d "$MOUNT_POINT" ]; then
+    echo "Error: Current working directory '$MOUNT_POINT' is not a valid directory."
+    exit 1
+fi
 
-# Generate a fixed keying identifier
-raw_key=$(_generate_raw_encryption_key)
-keyid=$(_add_enckey $testdir "$raw_key" | awk '{print $NF}')
+# Create the test directory
+test_dir="$MOUNT_POINT/$test_dir"
+mkdir "$test_dir"
+
+# Check the fscrypt status for the mount point
+STATUS_OUTPUT=$(sudo $FSCRYPT_CLI status "$MOUNT_POINT" --verbose || true)
+if echo "$STATUS_OUTPUT" | grep -q "users can create fscrypt metadata on this filesystem"; then
+    echo "The mount point '$MOUNT_POINT' is already initialized with fscrypt."
+else
+    echo "The mount point '$MOUNT_POINT' is not properly initialized. Initializing now..."
+
+    # Initialize fscrypt on the mount point
+    if sudo $FSCRYPT_CLI setup "$MOUNT_POINT" --verbose --force --all-users; then
+        echo "Successfully initialized fscrypt on '$MOUNT_POINT'."
+    else
+        echo "Error: Failed to initialize fscrypt on '$MOUNT_POINT'."
+        exit 1
+    fi
+fi
 
-case ${fscrypt} in
-	"none")
-		# do nothing for the test directory and will test it
-		# as one non-encrypted directory.
-		pushd $testdir
-		${mydir}/../suites/${testcase}.sh
-		popd
-		clean_up
-		;;
-	"unlocked")
-		# set encrypt policy with the key provided and then
-		# the test directory will be encrypted & unlocked
-		_set_encpolicy $testdir $keyid
-		pushd $testdir
-		${mydir}/../suites/${testcase}.sh
-		popd
-		clean_up
-		;;
-	"locked")
-		# remove the key, then the test directory will be locked
-		# and any modification will be denied by requiring the key
-		_rm_enckey $testdir $keyid
-		clean_up
-		;;
-	*)
-		clean_up
-		echo "Unknown parameter $1"
-		exit 1
-esac
+case ${fscrypt_type} in
+    "none")
+        # Test non-encrypted directory
+        pushd "$test_dir"
+        "${mydir}/../suites/${test_case}.sh"
+        popd
+        ;;
+    "unlocked")
+        # Encrypt but leave unlocked the test directory
+		printf "password\npassword\n" | sudo $FSCRYPT_CLI encrypt "$test_dir" --verbose --source=custom_passphrase  --name="test_secret"
+        pushd "$test_dir"
+        "${mydir}/../suites/${test_case}.sh"
+        popd
+        ;;
+    "locked")
+        # Encrypt and lock the directory
+        printf "password\npassword\n" | sudo $FSCRYPT_CLI encrypt "$test_dir" --verbose --source=custom_passphrase  --name="test_secret"
```

Comment:
You can use the flag `--skip-unlock` during encrypt and skip the next line.

Reply:
will try it